Automattic/wp-calypso
# Launchpad - choose a domain task launchpad data fetching update

Related to https://github.com/Automattic/wp-calypso/issues/73830

## Time Estimate

Review: Short
Test: Mid

## Proposed Changes

* Using `useLaunchpad` hook to fetch launchpad related data and use it for the `/domains/add/{siteSlug}?domainAndPlanPackage=true` flow

## Testing Instructions



* Checkout this branch
* Create a new launchpad enabled site through general onboarding (`/start/`) 
* Open the browser console and watch for any related errors.
* Click the `Choose a domain` task on the launchpad screen
* Verify the `Next Steps` back navigation redirects to the launchpad screen
* Verify the "What's next?" `Next Steps` button redirects to the launchpad screen

![image](https://user-images.githubusercontent.com/10482592/225148345-d7570334-4af2-4126-8590-14b9d01b4bcd.png)


![image](https://user-images.githubusercontent.com/10482592/225148272-25e940b9-8cfb-44fd-83e6-53d2157a4b25.png)


## Pre-merge Checklist
